### PR TITLE
tests: fix 02305_schema_inference_with_globs flakiness

### PR DESCRIPTION
**Pull request by azat**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
tests: fix 02305_schema_inference_with_globs flakiness

Order of files on disk is not guarantee to match with:
- order of creation
- lexical order

So sometimes 02305_data4.jsonl comes first and 2 rows is enough to get
schema.

Reorganize checks a little to avoid flakiness.

Also make it possible to run this test in parallel.

Refs: #38337 (cc @Avogar )

### CODE REVIEW

**Review by serxa — APPROVED** [review]

**Comment by Avogar** [comment]
@azat Couldn't find failed test 02305_schema_inference_with_globs after https://github.com/ClickHouse/ClickHouse/pull/38337, can you share where did you see it?

**Comment by azat** [comment]
>@azat Couldn't find failed test 02305_schema_inference_with_globs after https://github.com/ClickHouse/ClickHouse/pull/38337, can you share where did you see it?

Locally (I've started looking into it before your fix, since I saw failures of this test in my PRs), and for example here - https://s3.amazonaws.com/clickhouse-test-reports/37852/6de7590af40e7b7a7d5e5ecab7b8207591b0e0bf/stateless_tests__release__wide_parts_enabled__actions_.html

You can see such reports here - https://play.clickhouse.com/play?user=play#c2VsZWN0IAogIHJlcG9ydF91cmwsIGNoZWNrX3N0YXJ0X3RpbWUsIGhlYWRfcmVmLCBiYXNlX3JlZgpmcm9tIGNoZWNrcwp3aGVyZQogIGNoZWNrX3N0YXJ0X3RpbWUgPj0gJzIwMjItMDYtMjAnIGFuZAogIGJhc2VfcmVmID0gJ21hc3RlcicgYW5kCiAgaGVhZF9yZWYgIT0gJzIyLjYnIGFuZAogIHRlc3RfbmFtZSBsaWtlICcwMjMwNV9zY2hlbWFfaW5mZXJlbmNlX3dpdGhfZ2xvYnMnIGFuZAogIHRlc3Rfc3RhdHVzIGluICgnRkFJTCcsICdGTEFLWScpCm9yZGVyIGJ5IGNoZWNrX3N0YXJ0X3RpbWU=

File: tests/queries/0_stateless/02305_schema_inference_with_globs.sh [hunk 1]
```diff
@@ -1,18 +1,17 @@
 #!/usr/bin/env bash
-# Tags: no-fasttest
+# Tags: no-fasttest, no-parallel
```
Comment (Avogar) [inline tests/queries/0_stateless/02305_schema_inference_with_globs.sh:2]: Guess we can remove `no-parallel` now after adding `CLICKHOUSE_TEST_UNIQUE_NAME`

Comment (azat) [reply tests/queries/0_stateless/02305_schema_inference_with_globs.sh:2]: Great catch, I thought that I've removed it in the second patch, but seems forgot.

**Comment by azat** [comment]
>[Stateless tests flaky check (address, actions) — fail: 0, passed: 100](https://s3.amazonaws.com/clickhouse-test-reports/38511/9f147bde390ea5dc23551c5e93ae2b3e84ac2c0a/stateless_tests_flaky_check__address__actions_.html)

Yes, now it is compatible with run in parallel.

**Review by Avogar — APPROVED** [review]

**Pull request merged**